frequency_calculator.py

```diff
@@ -0,0 +1,36 @@
+from pathlib import Path
+from typing import Annotated
+
+import typer
+from typing_extensions import Optional
+
+from characters_counter import count_character_frequency
+from logging_config import setup_logging
+
+app = typer.Typer()
+logger = setup_logging()
+
+
+@app.command()
+def calculate(
+    file: Annotated[Optional[Path], typer.Option("--file", "-f")] = None,
+    paragraph: Annotated[Optional[str], typer.Option("--paragraph", "-p")] = None,
+) -> None:
+    if file and paragraph:
+        logger.warning("Warning! Both arguments is provided. Application will prioritize file.")
+        res = count_character_frequency(file)
+        logger.info(res)
+    elif file:
+        logger.debug("Calculating characters in file...")
+        res = count_character_frequency(file)
+        logger.info(res)
+    elif paragraph:
+        logger.debug("Calculating characters in paragraph...")
+        res = count_character_frequency(paragraph)
+        logger.info(res)
```

Comment:
I see a lot of repeated code with logger.info

Reply:
I agree with you. Thank you for pointing out me a guard clause pattern. I used it and it helped to reduce a repeated code.